src/jvm/main/org/jetbrains/kotlinx/lincheck/strategy/managed/ManagedStrategy.kt

```diff
@@ -937,6 +933,12 @@ abstract class ManagedStrategy(
             if (owner == null && atomicMethodDescriptor == null && guarantee == null) { // static method
                 LincheckJavaAgent.ensureClassHierarchyIsTransformed(className.canonicalClassName)
             }
+            if (atomicMethodDescriptor != null && atomicMethodDescriptor.kind.isSetter) {
```

Comment:
Should the `isCasSetter` case also be covered here?

Reply:
`isSetter` includes `isCasSetter`:

```kotlin
internal val AtomicMethodKind.isSetter get() = when (this) {
    SET,
    GET_AND_SET,
    COMPARE_AND_SET,
    WEAK_COMPARE_AND_SET,
    COMPARE_AND_EXCHANGE
         -> true
    else -> false
}

internal val AtomicMethodKind.isCasSetter get() = when (this) {
    COMPARE_AND_SET,
    WEAK_COMPARE_AND_SET,
    COMPARE_AND_EXCHANGE
         -> true
    else -> false
}
```

Would you prefer to split the two, so that `isSetter` covers only `SET` and `GET_AND_SET` ?

Reply:
Got it, thanks. I think this is fine.